### PR TITLE
Workspace: Inherit box-sizing from design system global styles

### LINKED ISSUE #6775: Design System: Update Box Sizing
## Feature Description

Throughout the editor and dashboard we're using border-box box-sizing. The DS has the default `content-box` intact. We should set global box-sizing at design system root for consistency. 

## Additional Context

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by samwhale**
## Context

We were developing in storybook without `border-box` sizing by default. All styles in the dashboard and editor use this sizing system, so this pr lines us up nice ✨.

## Summary

This PR does a few things:
- Updates storybook to use `box-sizing: border-box;` by default. This will make sure we develop components using the same box sizing as the dashboard and editor.
- Centralizes the box-sizing styles to one place: the design system.
- Fixes toggle styling (using border-box sizing now)
- Fixes cursor on checkbox

## Relevant Technical Choices

- Created a new global style called `Styles` which will have all the global styling in one place

## To-do

After this is merged: need to remove these changes: https://github.com/google/web-stories-wp/pull/6759/files#diff-47c08de7f287efa529a3ddc68420bfda8f7edf0b8cd8ebde09f417622ce2dadbR42-R78. We don't need those after the box-sizing is added here.

## User-facing changes

Only one change should be visible in the app: the toggle shouldn't look weird anymore.

|Before|After|
|--|--|
|image|image|
|image|image|

## Testing Instructions

Browse around the app, interact with form elements and buttons. Everything should look the same except the toggle. Sorry, I wish I had better testing instructions 😓.

### QA

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1. Browse around the app, interact with form elements and buttons. Everything should look the same except the toggle. Sorry, I wish I had better testing instructions 😓. This change touches everything but nothing (except the toggle) should look different.

### UAT

- [x] UAT should use the same steps as above.

### Does this PR have a security-related impact?

no

### Does this PR change what data or activity we track or use?

no

### Does this PR have a legal-related impact?

no

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testiing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #6775

### CODE REVIEW

**Comment by samwhale** [comment]
There's two files left that have `box-sizing: border-box` but wasn't sure if I could remove those:
1. https://github.com/google/web-stories-wp/blob/main/assets/src/edit-story/components/moveable/cropStyle.js#L31
    This global style doesn't look like it's being used anywhere...but it's still scary since there's an `!important` flag 😬 
2. https://github.com/google/web-stories-wp/blob/main/packages/activation-notice/src/app/components/messageContent.js#L58
    This is part of the `packages` folder. Not sure if it's okay to remove this or not 😓

File: assets/src/design-system/components/checkbox/index.js [hunk 1]
```diff
@@ -66,6 +66,7 @@ const CheckboxContainer = styled.div(
       margin: 0;
       padding: 0;
       opacity: 0;
+      cursor: pointer;
```
Comment (BrittanyIRL) [inline assets/src/design-system/components/checkbox/index.js:69]: Sorry, this was me, it seemed like it was working fine. 🤷‍♀️

File: assets/src/design-system/theme/global/styles.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+/**
+ * Internal dependencies
+ */
+import { overrideFocusOutlineStyles } from './overrideFocusOutline';
+
+/**
+ * All components in the design system are created using the following global styling.
+ * Not having these css rules may result in styling defects.
+ */
+export const Styles = createGlobalStyle`
```
Comment (BrittanyIRL) [inline assets/src/design-system/theme/global/styles.js:30]: 👏

File: docs/design-system.md [hunk 3]
```diff
@@ -17,7 +17,7 @@ The design system contains a full new color palette and theme. The structure is
...
 2. The new theme provides a dark and a light variant. Dark is the default. All colors are available regardless of variant, but the `theme.colors.bg` and `theme.colors.fg` are what you should gravitate towards using because the light and dark theme colors mirror each other. This will make dark and light mode functionality easier to handle down the road.
 3. Subdirectory called [helpers](../assets/src/design-system/theme/helpers/index.js) - these are styles meant to be quick references for one off components in the editor and dashboard that aren’t in the design system. They’re also used in the design system for continuity.
-4. Subdirectory called [global](../assets/src/design-system/theme/global/index.js) - these are exported as `ThemeGlobals` and should be added to the app root to handle global styles for both the dashboard and editor. Currently only the focus override exists here.
+4. Subdirectory called [global](../assets/src/design-system/theme/global/index.js) - these are exported as `ThemeGlobals` and should be added to the app root to handle global styles for both the dashboard and editor.
```
Comment (BrittanyIRL) [inline docs/design-system.md:20]: yay, you updated the docs!!! ❤️

**Comment by miina** [comment]
> https://github.com/google/web-stories-wp/blob/main/assets/src/edit-story/components/moveable/cropStyle.js#L31
This global style doesn't look like it's being used anywhere...but it's still scary since there's an !important flag 😬

@samwhale This global style is used in the media edit mode for the resize handlers -- it's overriding the values from the Moveable library we're using. It looks like it can be safely removed now though (let's just review the related Percy screenshots more carefully just in case).

File: assets/src/edit-story/integrations/wordpress/metaBoxes/metaBoxesArea.js [hunk 4]
```diff
@@ -43,11 +43,6 @@ const Wrapper = styled.div`
     box-sizing: content-box;
   }
 
-  textarea,
-  input {
-    box-sizing: border-box;
-  }
-
```
Comment (barklund) [inline assets/src/edit-story/integrations/wordpress/metaBoxes/metaBoxesArea.js:50]: Are we sure these aren't needed anymore? Are they still children of a relevant (fixed) node or are they somewhere outside the tree? @swissspidy?

**Review by maxyinger — APPROVED** [review]

File: assets/src/edit-story/integrations/wordpress/metaBoxes/metaBoxesArea.js (hunk 4 above)
Comment (swissspidy) [reply assets/src/edit-story/integrations/wordpress/metaBoxes/metaBoxesArea.js:50]: Would need to double check by toggling the meta box area and see if everything is still in order.

**Review by BrittanyIRL — APPROVED** [review]
👍

File: assets/src/edit-story/integrations/wordpress/metaBoxes/metaBoxesArea.js (hunk 4 above)
Comment (samwhale) [reply assets/src/edit-story/integrations/wordpress/metaBoxes/metaBoxesArea.js:50]: Added the `All in One SEO Pack` plugin. There's a lot of stuff there but it is getting the box sizing styles: 
Screen Shot 2021-03-17 at 9 44 22 AM

**Comment by samwhale** [comment]
Editor integration tests are failing for unrelated reasons to this pr (failing on main too). The dropdown fixture `optionList` can't find the `listbox` role on the container of the buttons in the new dropdown.

**Comment by BrittanyIRL** [comment]
> Editor integration tests are failing for unrelated reasons to this pr (failing on main too). The dropdown fixture `optionList` can't find the `listbox` role on the container of the buttons in the new dropdown.

Since they're failing on main too, can you make a new ticket to fix this?

**Comment by samwhale** [comment]
> > Editor integration tests are failing for unrelated reasons to this pr (failing on main too). The dropdown fixture `optionList` can't find the `listbox` role on the container of the buttons in the new dropdown.
> 
> Since they're failing on main too, can you make a new ticket to fix this?

Added here: #6937

**Comment by miina** [comment]
Hm, how did the failing test slip through to the `main`, I wonder.

We should not merge failing tests btw, if there's a flaky test that has been confirmed to be definitely unrelated then let's `xit` these tests and create an issue for fixing the flaky test. Merging failing tests that seem unrelated can be a bit dangerous.

cc @swissspidy Should we perhaps turn back on not allowing merges with failing tests?

**Comment by swissspidy** [comment]
Please see #6937. And also https://xwp.slack.com/archives/CCCAUAH6F/p1615565503065600?thread_ts=1615479664.056100&cid=CCCAUAH6F, as discussed on Slack.

**Comment by samwhale** [comment]
> Hm, how did the failing test slip through to the `main`, I wonder.
> 
> We should not merge failing tests btw, if there's a flaky test that has been confirmed to be definitely unrelated then let's `xit` these tests and create an issue for fixing the flaky test. Merging failing tests that seem unrelated can be a bit dangerous.


Sorry about that, I'll make sure to add `xit` to tests in this scenario in the future.

**Pull request merged**